### PR TITLE
fix: Possible use of a wrong region in s3 buckets

### PR DESCRIPTION
**Pull request by irmatov**

### CODE REVIEW

**Review by disq — APPROVED** [review]
Good catch! but I have a nit (probably the linter would also catch it)

File: resources/services/s3/buckets.go [hunk 1]
```diff
@@ -513,9 +513,11 @@ func fetchS3BucketsWorker(ctx context.Context, meta schema.ClientMeta, buckets <
 	for bucket := range buckets {
 		// always set default bucket region to us-east-1
 		wb := &WrappedBucket{Bucket: bucket, Region: "us-east-1"}
-		e := resolveS3BucketsAttributes(ctx, meta, wb)
-		res <- wb
-		errs <- e
+		if err := resolveS3BucketsAttributes(ctx, meta, wb); err == nil {
```
[the remark this answers is not in the document]
Comment (irmatov) [reply resources/services/s3/buckets.go]: Updated as suggested.

**Review by roneli — APPROVED** [review]
Nice! good catch indeed.

**Comment by disq** [comment]
Took a look at the source, wouldn't fix not found errors being reported as us-east-1. Because in resolveS3BucketsAttributes:

```go
// call to GetBucketRegion()
		if c.IsNotFoundError(err) {
			return nil
		}
```

Then we get that nil insert the bucket. Maybe we need a proxy error or a `processed bool` flag?

**Comment by irmatov** [comment]
@disq Updated. Moved checking for NotFound higher up the stack.

**Pull request merged**